core/src/tangram.cpp

```diff
@@ -90,16 +98,21 @@ static std::bitset<8> g_flags = 0;
 Map::Map() {
 
     impl.reset(new Impl());
+    impl->valid.store(true);
 
 }
 
 Map::~Map() {
     // The unique_ptr to Impl will be automatically destroyed when Map is destroyed.
+    impl->valid.store(false);
     TextDisplay::Instance().deinit();
     Primitives::deinit();
 }
 
 void Map::Impl::setScene(std::shared_ptr<Scene>& _scene) {
+
+    if (!valid.load()) { return; }
+
     {
```

Comment:
Hmm Partially agree with this.
1. `valid` is atomic so should not race.
2. different implementations of unique_ptr makes `impl` valid or invalid. Its not guaranteed to be invalid because of that, as we saw yesterday.